### PR TITLE
Fix: MUMPS source is regularly down use the mirror

### PR DESCRIPTION
**Pull request by senhalil**
per [this comment](https://github.com/coin-or/Cbc/issues/529#issuecomment-1200666236)

### CODE REVIEW

**Comment by tkralphs** [comment]
It is not clear to me why the test is failing now when I think it passed previously and nothing substantive is changed with this PR. There must be bit rot of some sort going on here, but I don't have the bandwidth to dig into it. @dpo any quick ideas? I guess we could merge anyway, since it should not make things any worse.

**Comment by dpo** [comment]
Here is the error:
https://github.com/coin-or-tools/homebrew-coinor/actions/runs/3090516467/jobs/5017063518#step:8:49
It seems like `libm` isn't found. I'm not sure for what reason.

This PR doesn't affect the bottles and could be merged anyways of course, but the root issue will persist.

FYI The main MUMPS server was down for most of the summer (while all of France was on vacation). There's an official mirror here:
```ruby
  mirror "https://graal.ens-lyon.fr/MUMPS/MUMPS_5.5.1.tar.gz"
```

**Comment by dpo** [comment]
@senhalil Does the formula build from source locally for you?

**Comment by tkralphs** [comment]
>FYI The main MUMPS server was down for most of the summer (while all of France was on vacation). There's an official mirror here:
>```
> mirror "https://graal.ens-lyon.fr/MUMPS/MUMPS_5.5.1.tar.gz"
> ```
Oh, never realized there is an "official mirror". Also, I take it that there is a way to specify a mirror in the recipe? So maybe this PR needs to be modified? I was wondering about that.

**Comment by tkralphs** [comment]
> It seems like libm isn't found. I'm not sure for what reason.
Ah, I see. I completely missed that when I was checking the output. Weird...

**Comment by dpo** [comment]
> > FYI The main MUMPS server was down for most of the summer (while all of France was on vacation). There's an official mirror here:
> > ```
> > mirror "https://graal.ens-lyon.fr/MUMPS/MUMPS_5.5.1.tar.gz"
> > ```
> 
> Oh, never realized there is an "official mirror". Also, I take it that there is a way to specify a mirror in the recipe? So maybe this PR needs to be modified? I was wondering about that.

Yes, just use `mirror` as I did above.

**Comment by tkralphs** [comment]
@senhalil could you make the suggested change? 

I can't make heads or tails out of that error. The compile command being executed doesn't even include `-lm`.

File: mumps-seq.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 class MumpsSeq < Formula
   desc "Sequential Sparse Direct Solver"
   homepage "http://mumps-solver.org"
-  url "http://mumps.enseeiht.fr/MUMPS_5.4.1.tar.gz"
+  url "http://coin-or-tools.github.io/ThirdParty-Mumps/MUMPS_5.4.1.tar.gz"
```
Comment (dpo) [inline mumps-seq.rb:4]:
```suggestion
  url "http://mumps.enseeiht.fr/MUMPS_5.4.1.tar.gz"
  mirror "https://graal.ens-lyon.fr/MUMPS/MUMPS_5.4.1.tar.gz"
```

File: mumps-seq.rb [hunk 2]
```diff
@@ -2,6 +2,7 @@ class MumpsSeq < Formula
   desc "Sequential Sparse Direct Solver"
   homepage "http://mumps-solver.org"
   url "http://mumps.enseeiht.fr/MUMPS_5.4.1.tar.gz"
+  mirror "https://graal.ens-lyon.fr/MUMPS/MUMPS_5.4.1.tar.gz"
   sha256 "93034a1a9fe0876307136dcde7e98e9086e199de76f1c47da822e7d4de987fa8"
```
Comment (dpo) [inline mumps-seq.rb:6]:
```suggestion
  sha256 "93034a1a9fe0876307136dcde7e98e9086e199de76f1c47da822e7d4de987fa8"
  revision 1
```

File: mumps-seq.rb [hunk 3]
```diff
@@ -2,7 +2,9 @@ class MumpsSeq < Formula
   desc "Sequential Sparse Direct Solver"
   homepage "http://mumps-solver.org"
   url "http://mumps.enseeiht.fr/MUMPS_5.4.1.tar.gz"
```
Comment (dpo) [inline mumps-seq.rb:4]:
```suggestion
  url "https://graal.ens-lyon.fr/MUMPS/MUMPS_5.4.1.tar.gz"
```

File: mumps-seq.rb [hunk 4]
```diff
@@ -2,7 +2,9 @@ class MumpsSeq < Formula
...
+  mirror "https://graal.ens-lyon.fr/MUMPS/MUMPS_5.4.1.tar.gz"
```
Comment (dpo) [inline mumps-seq.rb:5]:
```suggestion
  mirror "http://mumps.enseeiht.fr/MUMPS_5.4.1.tar.gz"
```

**Comment by senhalil** [comment]
> @senhalil Does the formula build from source locally for you?

Yes, it did without further issues after the URL change.

**Comment by senhalil** [comment]
> Yes, just use `mirror` as I did above.

Thanks for this info, I wanted to add it as a mirror but didn't know how and didn't have the bandwidth to figure out so I hacked it by replacing the URL on local so PR reflects my ignorance ^^

**Comment by senhalil** [comment]
> @senhalil could you make the suggested change?

It looks like @dpo made the necessary changes.

However, the [coin-or mirror](http://coin-or-tools.github.io/ThirdParty-Mumps/MUMPS_5.4.1.tar.gz) is dropped during the changes. My suggestion would be keep it as an additional mirror (if it is possible to have multiple mirrors) or as the only mirror, just in case both of those links  go down at the same time as they are both in France.

**Comment by dpo** [comment]
MUMPS 5.5.1 is available now. Would you mind updating the formula?

**Comment by tkralphs** [comment]
I'm a bit at a loss as to what is causing this error. There doesn't seem to be a way to re-run workflows to see if the issue has ben fixed, but I guess if you update the recipe, the workflow will be run again. Let's see what happens. I don't have a strong feelig about adding our mirror versus just the official one. I will defer to @dpo .

**Comment by senhalil** [comment]
Sure, I'll do it first thing tomorrow.

**Comment by dpo** [comment]
fixed in #83

**Pull request closed without merging**